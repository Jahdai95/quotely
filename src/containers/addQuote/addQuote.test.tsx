import { render, screen, fireEvent } from "@testing-library/react";
import { describe, test, vi, expect, it } from "vitest";
import userEvent from "@testing-library/user-event";

import AddQuote from "./addQuote";
import { QuotesProvider } from "../../providers/quotes.provider";

describe("Should render AddQuote component", () => {
  const getRender = () => {
    return render(
      <QuotesProvider>
        <AddQuote open handleClose={() => vi.fn()} />
      </QuotesProvider>
    );
  };
  test("Should to add quote", async () => {
    getRender();

    const input = screen.getByTestId("input-add-quote");
    await userEvent.click(input);
    fireEvent.change(input, {
      target: { value: "La inteligencia no es sabiduría." },
    });
    expect(input).toHaveValue("La inteligencia no es sabiduría.");

    const button = screen.getByText("Guardar");
    await userEvent.click(button);
  });

  it("Should macth snapshot", () => {
    const { container } = getRender();
    expect(container).toMatchSnapshot();
  });
});
