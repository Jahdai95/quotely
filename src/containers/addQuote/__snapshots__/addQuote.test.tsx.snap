// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render AddQuote component > Should macth snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
